### DISCUSSION TITLE
schedules are not getting deleted when user code is deleted

### DISCUSSION THREAD

**Discussion by priyanka043** [zwave-js/zwave-js-ui#3483] [Q&A]
Hi,

Just want to ask that, when we delete the user code, does that doesn't delete the schedules that were created with respect to its user id?

**Comment by robertsLando** [comment]
Not sure if that should happen automatically @AlCalzone

**Reply by AlCalzone** [reply]
Not sure... I can see where the question is coming from, but on the other hand this could be unexpected.

**Reply by robertsLando** [reply]
So what?

**Reply by AlCalzone** [reply]
I don't remember the logic - do we only offer the schedules when the user code is enabled or always? If this is tied to the user ID status, it could make sense to delete the schedules when deleting a code.

**Reply by robertsLando** [reply]
There is not really a "delete user code" I watch for, I only watch for `userIdStatus` changes and based on that I update `available`/`enabled` array of user codes. 

https://github.com/zwave-js/zwave-js-ui/blob/99f26e5623857053ad7faf302880d80e02a52c89/api/lib/ZwaveClient.ts#L262-L294

**Reply by priyanka043** [reply]
yes, here delete user code is nothing but making the usercode available.
